test/e2e/suite_test.go

```diff
@@ -518,6 +520,45 @@ func createSingleDatacenterCluster(t *testing.T, ctx context.Context, namespace
 		return cassandraDatacenterReady(kdcStatus.Cassandra)
 	}, polling.k8ssandraClusterStatus.timeout, polling.k8ssandraClusterStatus.interval, "timed out waiting for K8ssandraCluster status to get updated")
 
+	// Confirm default images are being applied as expected.
+	expectedSystemLoggerContainerImage := &images.Image{
```

Comment:
It probably would be sufficient to test this via unit tests. Granted some form of coverage is better than no coverage.

Reply:
I looked some more at the test update, and it does feel out of place in the e2e test. What about adding/updating tests in `datacenter_test.go`?

Reply:
Are you saying to use an envtest instead? That's possible, but I don't think we can do it via a unit test because the `default` struct tags are acted on somewhere in the API server as far as I can tell.

Reply:
I initially wanted it in `k8ssandracluster_types_test.go` but there's no API server there.

Reply:
I am in fact suggesting unit tests, even if less than ideal :) In the tests we would have to copy from the annotations what the expected defaults are. That's a tradeoff I can live with. 

If you look at the what was done for the Reaper image in `pkg/reaper/deployment.go` the image defaults are defined as constants.

Reply:
It **sounds** like that would make my life a lot easier, so I'm very supportive of a plan in this direction :)  But I'm not 100% clear on what you're proposing? 

Can you fill out what you're thinking the unit under test is, the desired inputs and outputs? Would this test just amount to another test of the DC/Cluster level merge logic in `datacenter.go`?

Reply:
One of the main reasons I'm doing a test against the API server here is just that I'm actually not sure how the `default` struct tag is processed. I'm assuming that the API server does this work, but I'm not certain. 

Equally, I'm making a big assumption that using this tag will actually result in merge-like behaviour (where the user can just set a single field in `Image` and all other fields will fall back to their defaults). But what if I'm wrong and setting `SystemLoggerContainerImage` to any `Image` doesn't result in a merge, but just an overwrite of the whole `Image` instance?

Reply:
> I'm making a big assumption that using this tag will actually result in merge-like behaviour (where the user can just set a single field in Image and all other fields will fall back to their defaults). But what if I'm wrong and setting SystemLoggerContainerImage to a partially definedImage doesn't result in a merge, but just an overwrite of the whole Image instance?

So I think I was right. Using the `default` tag doesn't appear to solve the issue as we were hoping. This is the part of the test which checks that patching with only a partial `Image` instance will result in the defaults being applied to the other fields. This is basically how it ends:
```
                                expected  : &images.Image{Registry:"docker.io", Repository:"k8ssandra", Name:"system-logger", Tag:"d94231ba", PullPolicy:"", PullSecretRef:(*v1.LocalObjectR
                                actual: &images.Image{Registry:"docker.io", Repository:"", Name:"", Tag:"d94231ba", PullPolicy:"", PullSecretRef:(*v1.LocalObjectReference)(nil)}
```

As we can see, it appears the image only has one field defined. Using the `default` struct tag will not avoid the issue I was concerned about where changing any part of the image (e.g. tag) meant you had to define every field in the image.